draft-ietf-radext-radiusdtls-bis.md

```diff
@@ -624,6 +624,30 @@ This practice causes increased complexity in the client application and increase
 RADIUS/DTLS clients SHOULD implement session resumption, preferably stateless session resumption as given in {{RFC5077}}.
 This practice lowers the time and effort required to start a DTLS session with a server and increases network responsiveness.
 
+# Implementation Guidelines
+
+The text above describes the protocol. In this section, we give additional implementation guidelines. These guidelines are not part of the protocol, but they may help implementors create simple, secure, and interoperable implementations.
+
+## Client Implementations
+
+RADIUS/(D)TLS clients should use connected sockets where possible. Use of connected sockets means that the underlying kernel tracks the sessions so that the client subsystem does not need to manage multiple sessions on one socket.
+
+RADIUS/(D)TLS clients should use a single source (IP + port) when sending packets to a particular RADIUS/(D)TLS server. Doing so minimizes the number of (D)TLS session setups. It also ensures that information about the home server state is discovered only once.
+
+In practice, this means that RADIUS/(D)TLS clients with multiple internal RADIUS sources should use a local proxy that arbitrates all RADIUS traffic between the client and all servers. The proxy should accept traffic only from the authorized subsystems on the client machine and should proxy that traffic to known servers. Each authorized subsystem should include an attribute that uniquely identifies that subsystem to the proxy so that the proxy can apply origin-specific proxy rules and security policies. We suggest using NAS-Identifier for this purpose.
```

Comment:
We already have some of this text in the "Client Subsystems" section in the Security considerations. Would you suggest merging the "Client Subsystems" section in this section?
Otherwise, this text should at least reference the security considerations.